File: Python/Programmars/Level_1/예산.py
def solution(d, budget):
    answer = 0
    total = 0
    d.sort()
    for x in d:
        if (total+x) <= budget:
            total += x
            answer += 1
    return answer


print(solution([1, 3, 2, 5, 4], 9))  # 3
print(solution([2, 2, 3, 3], 10))  # 10

File: JavaScript/Programmars/썸머윈터 코딩/예산.js
function solution(d, budget) {
  let answer = 0;
  let rest = budget;
  d.sort((a, b) => a - b);

  for (let x of d) {
    if (x <= rest) {
      rest -= x;
      answer++;
    }
  }

  return answer;
}

console.log(solution([1, 3, 2, 5, 4], 9)); // 3
console.log(solution([2, 2, 3, 3], 10)); // 4
